### PR TITLE
Make optimizer not complain about parameters with requires_grad=False (#7338)

### PR DESCRIPTION
**Pull request by domaala**
This makes optimizer not complain about the parameters that do not require gradients (`requires_grad=False`), as they don't affect it anyway. This allows for easier and cleaner construction of an optimizer without having to prefilter the parameters. Referenced and discussed in #7338 .

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Comment by soumith** [comment]
thanks @domaala !
